src/arkode/arkode_mristep.c

```diff
@@ -2828,6 +4487,19 @@ int mriStepInnerStepper_HasRequiredOps(MRIStepInnerStepper stepper)
   else { return ARK_ILL_INPUT; }
 }
 
+/* Check whether stepper supports fast/slow tolerance adaptivity */
+sunbooleantype mriStepInnerStepper_SupportsRTolAdaptivity(MRIStepInnerStepper stepper)
+{
+  if (stepper == NULL) { return SUNFALSE; }
+  if (stepper->ops == NULL) { return SUNFALSE; }
+
+  if (stepper->ops->geterror && stepper->ops->reseterror && stepper->ops->setrtol)
+  {
+    return SUNTRUE;
+  }
+  else { return SUNFALSE; }
+}
+
```

Comment:
@Steven-Roberts I missed this in the `SUNStepper` PR but should `mriStepInnerStepper_EvolveSUNStepper` below call `SUNStepper_SetForcing`?

Reply:
Yes, good catch. SUNSteppers don't clear forcing after an evolve, but MRI is expected to do that.

Reply:
I think it also needs the `Set` call before the evolve to attach the forcing data since MRIStep expects the InnerStepper to handle this too

Reply:
@Steven-Roberts, can you insert the correct `SetForcing` calls in here that should have been in the SUNStepper PR?  either a "suggestion" or just a direct commit would be fine.

Reply:
Yes, I can make a commit this afternoon

Reply:
Done. Here's the diff: https://github.com/LLNL/sundials/pull/564/commits/8ce81d87650026ae52ed790c2ff7ed1adf40fb14